### PR TITLE
feat: issuer micorservice for issuing W3C credentials to aries-vcr

### PR DESCRIPTION
**Pull request by amanji**

### CODE REVIEW

File: routers/credential.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    except Exception as e:
+        # TODO: Log the error
+        print(e)
+        return Response(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            content=json.dumps(
+                {"error": "There was an error issuing the credential"}
+            ),
+            media_type="application/json",
+        )
```
Comment (PatStLouis) [inline routers/credential.py:40]: I like to import the `HTTPException` class from fastapi and raise an error for error handling. You can also raise an error from a package that isn't directly returning a Response.

```python
from fastapi import HTTPException
[...]
raise HTTPException(status_code=400, detail="There was an error issuing the credential.")
```



Another tip I like to use is to return a `JSONResponse` object if its not an error.

```python
from fastapi.responses import JSONResponse
[...]
return JSONResponse(status_code=200, content={})
```

File: schemas/credential_type.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    """CredentialType schema"""
+
+    verification_methods: Annotated[
+        list[str],
+        Body(
+            validation_alias="verificationMethods",
+            serialization_alias="verificationMethods",
+        ),
+    ]
+    oca_bundle: Annotated[dict | None, Body(validation_alias="ocaBundle")] = None
```
Comment (PatStLouis) [inline schemas/credential_type.py:18]: I think a pydantic Field could be useful here:

```python
from pydantic import BaseModel, Field
[...]
    oca_bundle: dict = Field(None, alias='ocaBundle')
```

File: schemas/secured_credential_with_options.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+class SecuredCredentialOptions(Options):
+    """SecuredCredentialOptions schema"""
+
+    credential_id: Annotated[
+        str,
+        Body(
+            validation_alias="credentialId",
+            serialization_alias="credentialId",
+        ),
+    ]
```
Comment (PatStLouis) [inline schemas/secured_credential_with_options.py:16]: same thing, could a pydantic Field have the same use here?
```python
from pydantic import Field
class SecuredCredentialOptions(Options):
    """SecuredCredentialOptions schema"""

    credential_id: str = Field(alias='credentialId')
```

File: routers/credential.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+@router.post(
+    "/",
+    status_code=status.HTTP_201_CREATED,
+    response_model=SecuredCredentialWithOptions,
+    response_model_exclude_none=True,
+    response_model_by_alias=True,
+)
+async def issue_credential(secured_credential: SecuredCredentialWithOptions):
+    """Issue a new credential"""
```
Comment (PatStLouis) [inline routers/credential.py:19]:
```suggestion
async def receive_credential(secured_credential: SecuredCredentialWithOptions):
    """Receive a new credential"""
```

The term issuing is misleading here as you are receiving already issued credentials

File: routers/credential.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        svc_response = await vcr_service.issue_credential(data)
+        # TODO: Log the response
+        print(svc_response)
+    except Exception as e:
+        # TODO: Log the error
+        print(e)
+        return Response(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            content=json.dumps(
+                {"error": "There was an error issuing the credential"}
```
Comment (PatStLouis) [inline routers/credential.py:37]:
```suggestion
                {"error": "There was an error storing the credential"}
```

File: schemas/secured_document.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
+from typing import Annotated, Generic, TypeVar
+from fastapi import Body
+from pydantic import BaseModel
+
+SecuredDocumentT = TypeVar("SecuredDocumentT")
```
Comment (PatStLouis) [inline schemas/secured_document.py:5]: Is the T at the end of `SecuredDocumentT` a typo?

Comment (amanji) [reply schemas/secured_document.py:5]: It's a generic, so the T indicates a type

File: schemas/credential_type.py (hunk 2 above)
Comment (amanji) [reply schemas/credential_type.py:18]: `Body` wraps `Field` from what I understand. It is provided by `fastapi` but is extended for request bodies

**Pull request merged**